sdk/identity/identity/test/node/managedIdentityCredential.spec.ts

```diff
@@ -191,6 +193,65 @@ describe("ManagedIdentityCredential", function() {
     }
   });
 
+  it("sends an authorization request correctly in an App Service environment", async () => {
+    // Trigger App Service behavior by setting environment variables
```

Comment:
Since the code for these two tests is identical to those above, maybe it'd be better to factor out the test code into a function that can be called with the environment variable names to use?  That way the two test cases would just be invoking the same shared function with different env var names as parameters.

Reply:
This is an interesting piece of feedback because we've been talking about not doing that kind of refactoring in tests.

Instead of factoring this code out, and since this PR isn't urgent, I think it's a good opportunity to review the guidelines.

Here's the PR: https://github.com/Azure/azure-sdk-for-js/pull/7784

Here's one of the quotes that could be used to argue your feedback:

> **Anti-pattern 1**, code that displays information (whether values or method names) that are absolutely related to the intention of the test should not be abstracted away [...]

From: https://github.com/Azure/azure-sdk-for-js/blob/5fea51e15c56060f926546afc2183204e5971259/documentation/writing_tests.md#shared-and-reusable-code

Let me know what you think! If you disagree we should ask others and change the proposed guidelines.

Reply:
After thinking about it for considerable time, I think this kind of automation should be allowed for internal tests. I'll do that change next week on the guidelines.